core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -136,8 +132,15 @@ void SYCL::impl_static_fence(const std::string& name) {
       [&]() {
         // guard accessing all_queues
         std::lock_guard<std::mutex> lock(Impl::SYCLInternal::mutex);
-        for (auto& queue : Impl::SYCLInternal::all_queues)
-          Impl::SYCLInternal::fence(**queue);
+        for (auto& queue : Impl::SYCLInternal::all_queues) {
```

Comment:
This only reports the first exception it finds.  I assume that is okay.

Reply:
Yes, I didn't change anything in the behavior here.